src/htmlminifier.js

```diff
@@ -941,336 +964,425 @@ function minify(value, options, partialMarkup) {
+                  buffer[charsIndex] = buffer[charsIndex].replace(/\s+$/, function(trailingSpaces) {
+                    text = trailingSpaces + text;
+                    return '';
+                  });
+                }
               }
-              if (buffer.length > 1 && (!prevComment || !options.conservativeCollapse && / $/.test(currentChars))) {
-                var charsIndex = buffer.length - 2;
-                buffer[charsIndex] = buffer[charsIndex].replace(/\s+$/, function(trailingSpaces) {
-                  text = trailingSpaces + text;
-                  return '';
-                });
+            }
+            if (prevTag) {
+              if (prevTag === '/nobr' || prevTag === 'wbr') {
+                if (/^\s/.test(text)) {
+                  var tagIndex = buffer.length - 1;
+                  while (tagIndex > 0 && buffer[tagIndex].lastIndexOf('<' + prevTag) !== 0) {
+                    tagIndex--;
+                  }
+                  trimTrailingWhitespace(tagIndex - 1, 'br');
+                }
               }
+              else if (inlineTextTags(prevTag.charAt(0) === '/' ? prevTag.slice(1) : prevTag)) {
+                text = collapseWhitespace(text, options, /(?:^|\s)$/.test(currentChars));
+              }
+            }
+            if (prevTag || nextTag) {
+              text = collapseWhitespaceSmart(text, prevTag, nextTag, options);
+            }
+            else {
+              text = collapseWhitespace(text, options, true, true);
+            }
+            if (!text && /\s$/.test(currentChars) && prevTag && prevTag.charAt(0) === '/') {
+              trimTrailingWhitespace(buffer.length - 1, nextTag);
             }
           }
-          if (prevTag) {
-            if (prevTag === '/nobr' || prevTag === 'wbr') {
-              if (/^\s/.test(text)) {
-                var tagIndex = buffer.length - 1;
-                while (tagIndex > 0 && buffer[tagIndex].lastIndexOf('<' + prevTag) !== 0) {
-                  tagIndex--;
+          if (!stackNoCollapseWhitespace.length && nextTag !== 'html' && !(prevTag && nextTag)) {
+            text = collapseWhitespace(text, options, false, false, true);
+          }
+        }
+        if (options.processScripts && specialContentTags(currentTag)) {
+          text = processScript(text, options, currentAttrs);
+        }
+
+        var runningAsyncTask = false;
+
+        /**
+         * To be called when running an async task.
+         *
+         * @param {AsyncTextPlaceholder} placeholder
+         */
+        function onAsyncTask(placeholder) {
+          buffer.push(placeholder);
+          runningAsyncTask = true;
+          asyncTasksWaitingFor++;
+        }
+
+        /**
+         * Returns the callback function for an async task.
+         *
+         * @param {AsyncTextPlaceholder} placeholder
+         * @returns {Function}
+         */
+        function getAsyncTaskCallback(placeholder) {
```

Comment:
Please move `var runningAsyncTask = true;` into here and remove the two references above to avoid allocation in the synchronous case.

Reply:
Done